ui/lib/apps/InstanceProfiling/pages/Detail.tsx

```diff
@@ -153,29 +119,75 @@ function ViewResultButton({ rec, t }) {
     )
   }
 
-  const DropdownButton = () => {
-    return (
-      <Dropdown.Button
-        disabled={rec.state !== taskState.Success}
-        overlay={menu}
-        onClick={() => openResult(`${rec.view_options[0]}`)}
-      >
-        {t(
-          `instance_profiling.detail.table.columns.selection.types.${rec.view_options[0]}`
-        )}
-      </Dropdown.Button>
-    )
+  return (
+    <Dropdown.Button
+      disabled={rec.state !== taskState.Success}
+      overlay={menu}
+      onClick={() => openResult(`${rec.view_options[0]}`, rec)}
+    >
+      {t(
+        `instance_profiling.detail.table.columns.selection.types.${rec.view_options[0]}`
+      )}
+    </Dropdown.Button>
+  )
+}
+
+const openResult = async (openAs: string, rec) => {
+  const isProtobuf = rec.raw_data_type === RawDataType.Protobuf
+  let token: string | undefined = await getActionToken(rec.id, 'single_view')
+  if (!token) {
+    return
+  }
+  let profileURL: string
+
+  switch (openAs) {
+    case ViewOptions.Download:
+      token = await getActionToken(rec.id, 'single_download')
```

Comment:
oh, sorry, I didn't notice this difference to get token, so the former logic is fine.